docs/the-new-architecture/pillars-fabric-components.md

```diff
@@ -26,7 +28,84 @@ To create a Fabric Component, we have to follow these steps:
 
 Once these steps are done, the Component is ready to be consumend by an app. Therefore, the guide shows how to add it to an app, leveraging the _autolinking_, and how to reference it from the JavaScript code.
 
-### Javascript Specification
+### Folder Setup
+
+The easiest way to create a Component is as a separate module we will then import as a dependency for our apps. This has several benefits, among which it pushes for code reuse, it keeps the components decoupled from the rest of the apps and it makes it easier to leverage some automation already in place.
+
+For this guide, we are going to create a Fabric Component that centers some text on the screen.
+
+Let's create a new folder at the same level of our app and let's call it `RTNCenteredText`.
+
+In this folder, we are going to create three subfolders: `js`, `ios` and `android`.
+
+The final result should look like this:
+
+<figure>
+  <img width="500" alt="Folder Structure for a Fabric Component" src="/docs/assets/NewArchitecture/AppFolderStructure.png"/>
+  <figcaption>Initial folder structure for a Fabric Component.</figcaption>
+</figure>
+
+### JavaScript Specification
+
+In the **New Architecture**, we decided to have a single source of truth for Components API and we decided that it has to be a typed dialect of JavaScript (either [Flow](https://flow.org/) or [TypeScript](https://www.typescriptlang.org/)). We need a typed dialect because the **Codegen** has to generate some code in C++, Objective-C++ and Java which are all strongly typed languages: therefore, it needs information on types.
+
+Another important aspect of the JavaScript specification is the file name. A Component filename has to end with the `NativeComponent.js` (or `jsx`, `ts`, `tsx`) suffix. The **Codegen** process will look up for files ending with that suffix to generate the required code.
```

Comment:
Oh, I misunderstood this the first time I read it -- I interpreted as just talking about file extensions, but I think you're saying they all have to end in `...NativeComponent` as well?

This is pretty important, I think it's worth highlighting it in an info box